writers/streamingbatchwriter/streamingbatchwriter.go

```diff
@@ -128,9 +133,9 @@ func (w *StreamingBatchWriter) Flush(_ context.Context) error {
 		w.migrateWorker.flush <- done
 		<-done
 	}
-	if w.deleteWorker != nil {
+	if w.deleteStaleWorker != nil {
```

Comment:
Do we also need a case for deleteRecord here?

Reply:
We do